### PR TITLE
charts: add to m4d extraArgs/extraEnvs

### PR DESCRIPTION
**Pull request by hunchback**
this PR addresses a subset of issue #317 

to enable future customization of the the manager to support various deployment modes I have added to values.yaml two expansion/customization options:
- extraArgs - add extra arguments to be passed to the manager container
- extraEnvs - pass extra environment variables for the manager container

the basic approach has been taken from what is generally accepted for helm charts such as is used for example in the elastic/elasticsearch helm chart https://github.com/elastic/helm-charts/blob/master/elasticsearch/values.yaml#L35 (many more similar examples exist)

to verify that there has been no degradation due to the change I ran:

```
aidan@AIDANSHRIBMAN-SDFUE6N:~/projects/the-mesh-for-data/charts$ helm install m4d m4d         --namespace m4d-system
NAME: m4d
LAST DEPLOYED: Wed Feb  3 10:32:57 2021
NAMESPACE: m4d-system
STATUS: deployed
REVISION: 1
TEST SUITE: None
```

### CODE REVIEW

File: charts/m4d/templates/manager.yaml [hunk 1]
```diff
@@ -405,13 +405,19 @@ spec:
       - args:
         - --enable-leader-election
         - --enable-all-controllers
```
Comment (roee88) [inline charts/m4d/templates/manager.yaml:407]: I think that this overrides the other flags.

**Review by roee88 — CHANGES_REQUESTED** [review]
I suggest to either template it better not always include --enable-all-controllers or to change the manager main.go to take the parameters differently (e.g., remove enable-all and make it implicit if no other enable flag os set or something else that makes sense).

**Comment by hunchback** [comment]
> I suggest to either template it better not always include --enable-all-controllers or to change the manager main.go to take the parameters differently (e.g., remove enable-all and make it implicit if no other enable flag os set or something else that makes sense).

will move these into the values.yaml so that they can be overridden

File: charts/m4d/templates/manager.yaml (hunk 1 above)
Comment (hunchback) [reply charts/m4d/templates/manager.yaml:407]: ok, will move them to values.yaml

**Comment by ronenkat** [comment]
@hunchback looks good, only need to add @roee88's input.

**Review by roee88 — APPROVED** [review]

**Pull request merged**